in_depth/Testing.rst

```diff
@@ -0,0 +1,72 @@
+.. _Testing :
+
+Testing IncludeOS
+=================
+
+To test IncludeOS as a kernel and it's features, we have written a number of tests
+that can be reused by anyone to test there work on IncludeOS. You will find a list
+of our tests in the `IncludeOS/test <https://github.com/includeos/IncludeOS/tree/master/test>`__ folder.
+
+.. note:: The following testing instructions only work on Linux at the moment.
+
+Unit Tests
+~~~~~~~~~~
+
+All the tests are grouped by categories as listed in the folder. The unit tests for each of the categories of tests are located under ``IncludeOS/test/<category>/unit/<test-name>.cpp``.
+
+To run all the unit tests simultaneously do:
```

Comment:
This does not actually run the tests. Only installs them. Ready for cmake.

Reply:
See here how they are actually run (before we started doing coverage at the same time)
https://github.com/includeos/IncludeOS/blob/3fa07fd5377043d660d46c17399643592323ec5f/Jenkinsfile

Reply:
Separated this into two sections as follows: 

```

To install and build the unit tests do:

::

  $ conan install IncludeOS/test -pr gcc-7.3.0-linux-x86_64 -s build_type=Debug
  $ . ./activate.sh

Then to run the unit tests in random order do;

::

   $ ctest --schedule-random


If you want to get the coverage report for the unit tests as well do:

::

  # build with coverage
  $ . ./activate.sh && cmake -DCOVERAGE=ON -DCODECOV_HTMLOUTPUTDIR=<your-coverage-dir> IncludeOS/test
  # build and test with coverage
  $ . ./activate.sh && make -j nproc coverage

After running the unit tests, you can then view the coverage report in html format
in the path you specified.

```
We can remove the section on coverage if it's not something we are concentrating on yet.